### ISSUE TITLE
Complete the TraceIdRation specification

### ISSUE THREAD

**Issue by carlosalberto** [open-telemetry/opentelemetry-specification#1826]
As part of the Sampling SIG (and as reported as part of #1819), `TraceIdRatio` is so far the preferred default sampler. However, related questions have [arose](https://github.com/open-telemetry/opentelemetry-specification/issues/1819#issuecomment-885217267), in order to complete such sampler specification:

1. Use sufficiently random trace IDs, specify how to evaluate the ratio test directly from TraceID bits
2. Propagate a new uniform random number in the range (0, 1] in the W3C tracestate to facilitate the ratio test
3. Specify that TraceIDs need not be "very random" but dictate the use of a standard hashing algorithm.

One detail to be aware of here is that some vendors may be providing their existing ids, so we have to make sure this works in all scenarios.

cc @jmacd @tedsuo @anuraaga

**Comment by yurishkuro** [comment]
> @carlosalberto wrote:
> One detail to be aware of here is that some vendors may be providing their existing ids, so we have to make sure this works in all scenarios.

I agree with this - we cannot rely on trace-id to be completely random.

Of course there's always an option to support all 3 solutions via configuration, because some of them may be more palatable / suitable for different sites. For example, in my company option 2 would be very difficult to deploy, as many services are overly sensitive (i.e. can demonstrate numeric evidence of perf regression) to any extra payloads on the wire.

**Comment by jmacd** [comment]
Re: sensitivity to extra payload

This was why my original proposal in https://github.com/open-telemetry/oteps/pull/168 used `traceparent`, because there we can get this support on-by-default for 3 bytes per context. 

To me, the hashing option is the most difficult to specify and I would eliminate it. I can accept either a random TraceID or a tracestate entry, supporting both is within reason just a lot harder to specify.

**Comment by ocelotl** [comment]
I asked Claude to review this issue and Claude suggests it be closed with this explanation:

> The three open questions (random trace IDs, tracestate propagation, hashing) were resolved via the Sampling SIG work tracked in #4012: OTEP 235 (consistent probability sampling threshold in tracestate via W3C Trace Context Level 2 Random flag) was merged, and the current trace/sdk.md now has a fully-specified Stable 'TraceIdRatioBased' sampler with algorithm requirements plus the composable 'ProbabilitySampler'. The TraceIdRatio spec is complete.

To guard against a mistaken suggestion, Claude then independently re-verified this conclusion:

> All 3 open Qs resolved in current trace/sdk.md: Stable TraceIdRatioBased (mandates deterministic hash) + composable ProbabilitySampler using W3C TraceContext L2 Random flag/56-bit randomness (#3924) and tracestate rv/th sub-keys (tracestate-probability-sampling.md); Sampling SIG #4012 & #1819 closed COMPLETED. No single closing PR ref'd #1826 but the work is verifiably present.

Claude evaluated this issue with high confidence that it can be closed.

Claude has classified this issue as resolved.

The intention of this comment is to make it easier for a human being to review this issue before making a decision.

**Comment by svrnm** [comment]
Closing this issue as it seems to be addressed, if this is incorrect please reopen or create a new one stating the details that are missing.
